platforms/core-configuration/declarative-dsl-core/src/main/kotlin/org/gradle/internal/declarativedsl/dom/resolution/DefaultDocumentResolutionContainer.kt

```diff
@@ -93,11 +94,18 @@ class DefaultDocumentResolutionContainer(
     private val propertyResolution: Map<PropertyNode, PropertyResolution>,
     private val valueFactoryResolution: Map<ValueFactoryNode, ValueFactoryResolution>
 ) : DocumentResolutionContainer {
-    override fun data(node: ElementNode): ElementResolution = elementResolution.getValue(node)
-    override fun data(node: PropertyNode): PropertyResolution = propertyResolution.getValue(node)
-    override fun data(node: DeclarativeDocument.DocumentNode.ErrorNode): ErrorResolution = ErrorResolution
-    override fun data(value: ValueFactoryNode): ValueFactoryResolution = valueFactoryResolution.getValue(value)
-    override fun data(value: DeclarativeDocument.ValueNode.LiteralValueNode): LiteralValueResolved = LiteralValueResolved(value.value)
+    override fun data(node: ElementNode): ElementResolution =
+        elementResolution.getValue(node)
+    override fun data(node: PropertyNode): PropertyResolution =
+        propertyResolution.getValue(node)
+    override fun data(node: DeclarativeDocument.DocumentNode.ErrorNode): ErrorResolution =
+        ErrorResolution
+    override fun data(value: ValueFactoryNode): ValueFactoryResolution =
+        valueFactoryResolution.getValue(value)
+    override fun data(value: DeclarativeDocument.ValueNode.LiteralValueNode): LiteralValueResolved =
+        LiteralValueResolved(value.value)
+    override fun data(value: DeclarativeDocument.ValueNode.NamedReferenceNode): NamedReferenceResolved =
+        NamedReferenceResolved(value.referenceName) // TODO: probably it shouldn't just be the name... or should it?
```

Comment:
This should not always be a successful resolution. If the named reference resolved to anything other than an enum constant origin (or not resolved at all), it should be an unsuccessful resolution result instead. See the logic for value factory resolution above.

Reply:
I've given it a try, please see in the next commit.